ansible/site.yml

```diff
@@ -0,0 +1,45 @@
+- hosts: all
+  sudo: true
+  environment: proxy_env
+  tasks:
+    - hostname: "name={{ hostname }}"
+    - template: src=templates/envvar.sh.j2 dest=/etc/profile.d/envvar.sh owner=root group=root mode=0755
+    - apt: "update_cache=yes name={{ item }}"
+      with_items:
+        - unzip
+        - vim-nox
+        - curl
+        - python-software-properties
+        - git 
+    - shell: creates=/usr/local/go/bin/go curl -L https://storage.googleapis.com/golang/go1.4.2.linux-amd64.tar.gz | tar -xvz -C /usr/local
```

Comment:
It will be nice to not have to download the packages, if they are already installed, as downloading packages adds significantly to the setup time.

Just wondering if ansible allows for it in some way. I was reading about ansible modules being idempotent (http://docs.ansible.com/playbooks_intro.html#tasks-list), but not sure if that is applicable here. Basically I am thinking we can start with a base box, and have ansible provision it once. Then package the box and upload it and use it as our box (without touching the ansible script itself).